src/components/avatar/index.tsx

```diff
@@ -155,176 +154,105 @@ export type AvatarProps = Pick<AccessibilityProps, 'accessibilityLabel'> &
     testID?: string;
   }>;
 
+interface Statics {
+  badgePosition: typeof BadgePosition;
+}
+
 /**
  * @description: Avatar component for displaying user profile images
  * @extends: TouchableOpacity, Image
  * @image: https://github.com/wix/react-native-ui-lib/blob/master/demo/showcase/Avatar/Avarat_1.png?raw=true, https://github.com/wix/react-native-ui-lib/blob/master/demo/showcase/Avatar/Avarat_2.png?raw=true
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/AvatarsScreen.tsx
  */
-class Avatar extends PureComponent<AvatarProps> {
-  static displayName = 'Avatar';
-
-  styles: ReturnType<typeof createStyles>;
-
-  constructor(props: AvatarProps) {
-    super(props);
-
-    this.styles = createStyles(props);
-
-    if (props.imageSource) {
+const Avatar = forwardRef<any, AvatarProps>((props: AvatarProps, ref: React.ForwardedRef<any>) => {
+  const themeProps = useThemeProps(props, 'Avatar');
+  const {
+    imageSource,
+    source,
+    size = 50,
+    labelColor = Colors.$textDefault,
+    badgeProps = {},
+    badgePosition = BadgePosition.TOP_RIGHT,
+    testID,
+    ribbonLabel,
+    customRibbon,
+    ribbonStyle,
+    ribbonLabelStyle,
+    animate = false,
+    imageStyle,
+    onImageLoadStart,
+    onImageLoadEnd,
+    onImageLoadError,
+    imageProps,
+    label,
+    name,
+    backgroundColor,
+    useAutoColors,
+    autoColorsConfig,
+    containerStyle,
+    onPress,
+    children
+  } = themeProps;
+  const {size: _badgeSize, borderWidth = 0} = badgeProps;
+  const badgeSize = _badgeSize || DEFAULT_BADGE_SIZE;
```

Comment:
Consider moving the default value to the destruction (more readable)

Reply:
It was there, this is actually a (not so readable :/) fix for a test, if the input is `0` the result should be `DEFAULT_BADGE_SIZE`, let me know if you have a better suggestion.

Reply:
I see, so let's keep it that way